### PR TITLE
custom_target: libfoo.so.1 is linkable

### PR DESCRIPTION
**Pull request by xclaesse**

### CODE REVIEW

File: mesonbuild/build.py [hunk 1]
```diff
@@ -2595,11 +2595,18 @@ def get_dep_outname(self, infilenames):
                 raise InvalidArguments('Substitution in depfile for custom_target that does not have an input file.')
             return self.depfile
 
+    def is_linkable_output(self, output: str) -> bool:
+        if any(output.endswith(i) for i in {'.a', '.dll', '.lib', '.so', '.dylib'}):
+            return True
+        # libfoo.so.X soname
+        if re.match('.*\.so(\.\d+)*$', output):
```
Comment (bruchar1) [inline mesonbuild/build.py:2602]:
```suggestion
        if re.search(r'\.so(\.\d+)*$', output):
```

File: mesonbuild/build.py [hunk 2]
```diff
@@ -2595,11 +2595,18 @@ def get_dep_outname(self, infilenames):
                 raise InvalidArguments('Substitution in depfile for custom_target that does not have an input file.')
             return self.depfile
 
+    def is_linkable_output(self, output: str) -> bool:
+        if any(output.endswith(i) for i in {'.a', '.dll', '.lib', '.so', '.dylib'}):
```
[the remark this answers is not in the document]
Comment (xclaesse) [reply mesonbuild/build.py]: I did not know that syntax, done. Thanks.

File: mesonbuild/build.py (hunk 1 above)
Comment (xclaesse) [reply mesonbuild/build.py]: Assuming it's a type to have dropped `.*`? Done.

Comment (xclaesse) [reply mesonbuild/build.py]: ah sorry, you changed match to search, makes sense.

File: mesonbuild/build.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (xclaesse) [reply mesonbuild/build.py]: Ok, better not change that if in doubt anyway, removed the commit.

**Comment by xclaesse** [comment]
@eli-schwartz any other objection?

**Review by eli-schwartz — APPROVED** [review]

**Pull request merged**